libavformat/dashenc_http.c

```diff
@@ -427,15 +427,24 @@ static connection *claim_connection(char *url, int need_new_connection) {
         av_log(NULL, AV_LOG_INFO, "No free connections so added one. Url: %s, tail: %d\n", url, connections_tail->nr);
     }
 
+    if (conn == NULL) {
+        return conn;
+    }
+
     if (need_new_connection && conn->opened) {
         conn->opened = 0;
         ff_format_io_close(conn->s, &conn->out);
     }
-
-    av_log(NULL, AV_LOG_INFO, "Claimed conn_id: %d, url: %s\n", conn_nr, url);
-    len = strlen(url) + 1;
-    conn->url = malloc(len);
-    av_strlcpy(conn->url, url, len);
+    conn->url = NULL;
+    if (url != NULL) {
```

Comment:
I think should check whether url is not null at the start of the method.
It's a requirement of this method that it is always called with the url.
If no url is supplied we can return NULL.